src/main/java/com/qrlo/qrloservicecore/handler/AuthHandler.java

```diff
@@ -30,37 +33,35 @@
 @Component
 public class AuthHandler {
     private final JwtTokenProvider jwtTokenProvider;
-    private final KakaoService kakaoService;
     private final UserService userService;
     private final EmailService emailService;
+    private final OAuthService oAuthService;
 
-    public AuthHandler(JwtTokenProvider jwtTokenProvider, KakaoService kakaoService, UserService userService, EmailService emailService) {
+    public AuthHandler(JwtTokenProvider jwtTokenProvider, UserService userService,
+                       EmailService emailService, OAuthService oAuthService) {
         this.jwtTokenProvider = jwtTokenProvider;
-        this.kakaoService = kakaoService;
         this.userService = userService;
         this.emailService = emailService;
+        this.oAuthService = oAuthService;
     }
 
     public Mono<ServerResponse> authenticate(ServerRequest request) {
         Mono<AuthRequest> authRequestMono = request
                 .bodyToMono(AuthRequest.class)
-                .cache()
-                .switchIfEmpty(Mono.defer(() -> Mono.error(new ResponseStatusException(HttpStatus.UNAUTHORIZED))));
+                .switchIfEmpty(Mono.defer(() -> Mono.error(new ResponseStatusException(HttpStatus.UNAUTHORIZED))))
+                .cache();
```

Comment:
This is not right... You should not authN authZ no matter what if authZ string provided